### ISSUE TITLE
`initialize`: config is empty or not given at all (ArgumentError) when generating the config file

### ISSUE THREAD

**Issue by brandoncordell** [uploadcare/uploadcare-rails#53]
MacOS Sierra 10.12.2
Ruby 2.4.0
Rails 5.1.0.rc1
Project: Brand new rails project

    rails g uploadcare_config
      Warning: no uploadcare.yml config file found.
      while we will use default settings please run
        rails g uploadcare_config
      to set up config file.

    /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/uploadcare-rails-1.1.0/lib/uploadcare/rails/settings.rb:39:in `initialize': config is empty or not given at all (ArgumentError)
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/uploadcare-rails-1.1.0/config/initializers/uploadcare.rb:24:in `new'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/uploadcare-rails-1.1.0/config/initializers/uploadcare.rb:24:in `<top (required)>'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.0.rc1/lib/active_support/dependencies.rb:286:in `load'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.0.rc1/lib/active_support/dependencies.rb:286:in `block in load'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.0.rc1/lib/active_support/dependencies.rb:258:in `load_dependency'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.0.rc1/lib/active_support/dependencies.rb:286:in `load'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/engine.rb:655:in `block in load_config_initializer'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.0.rc1/lib/active_support/notifications.rb:166:in `instrument'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/engine.rb:654:in `load_config_initializer'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/engine.rb:612:in `block (2 levels) in <class:Engine>'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/engine.rb:611:in `each'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/engine.rb:611:in `block in <class:Engine>'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/initializable.rb:30:in `instance_exec'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/initializable.rb:30:in `run'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/initializable.rb:59:in `block in run_initializers'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:228:in `block in tsort_each'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:422:in `block (2 levels) in each_strongly_connected_component_from'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:431:in `each_strongly_connected_component_from'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:421:in `block in each_strongly_connected_component_from'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/initializable.rb:48:in `each'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/initializable.rb:48:in `tsort_each_child'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:415:in `call'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:415:in `each_strongly_connected_component_from'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:349:in `block in each_strongly_connected_component'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:347:in `each'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:347:in `call'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:347:in `each_strongly_connected_component'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:226:in `tsort_each'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/tsort.rb:205:in `tsort_each'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/initializable.rb:58:in `run_initializers'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.0.rc1/lib/rails/application.rb:353:in `initialize!'
    from /Users/brandoncordell/Code/uc_testing/config/environment.rb:5:in `<top (required)>'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/spring-2.0.1/lib/spring/application.rb:92:in `require'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/spring-2.0.1/lib/spring/application.rb:92:in `preload'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/spring-2.0.1/lib/spring/application.rb:143:in `serve'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/spring-2.0.1/lib/spring/application.rb:131:in `block in run'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/spring-2.0.1/lib/spring/application.rb:125:in `loop'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/spring-2.0.1/lib/spring/application.rb:125:in `run'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/spring-2.0.1/lib/spring/application/boot.rb:19:in `<top (required)>'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /Users/brandoncordell/.rbenv/versions/2.4.0/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from -e:1:in `<main>'

**Comment by armchairdj** [comment]
To put this in plain English: It won't let you generate a yml file without already having a yml file.

**Comment by vizvamitra** [comment]
@armchairdj yeah, this is a very annoying bug. I git blame a bit and it seems that it was introduced at 19 May 2016. This is shocking.

The library was abandoned for nearly a year, but this should change as I'm going to make some effort on improving it. I'll fix this bug shortly

**Comment by armchairdj** [comment]
Thanks for this patch! I just sent you a PR for some unrelated stuff. Would love to contribute more - particularly to documentation - if y'all would like some assistance.
